src/benchmarks/micro/corefx/System.Net.NetworkInformation/NetworkInterfaceTests.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Reflection;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Net.NetworkInformation.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX)]
+    public class NetworkInterfaceTests
+    {
+        private PropertyInfo[] NetworkInterfaceInfo = typeof(System.Net.NetworkInformation.NetworkInterface).GetProperties(BindingFlags.Instance | BindingFlags.Public);
+        private PropertyInfo[] IPInterfaceStatisticsInfo = typeof(System.Net.NetworkInformation.IPInterfaceStatistics).GetProperties(BindingFlags.Instance | BindingFlags.Public);
+        private PropertyInfo[] IPInterfacePropertiesInfo = typeof(System.Net.NetworkInformation.IPInterfaceProperties).GetProperties(BindingFlags.Instance | BindingFlags.Public);
```

Comment:
I think we should be explicit about the properties we're accessing, ideally with one benchmark per property we care about, rather than using reflection to enumerate all of them as part of a single test.

Reply:
I agree with Stephen, this should be explicit without reflection usage. Reflection can add some performance overhead and definitely introduce boxing which would be visible in the "allocated" memory column.

Example:

```cs
public long GetAllNetworkInterfacesProperties()
{
    long consume = 0;

    foreach (NetworkInterface ni in NetworkInterface.GetAllNetworkInterfaces())
    {
        consume += ni.Id.Length;
        consume += ni.SupportsMulticast ? 1 : 0;

        IPInterfaceStatistics ips = ni.GetIPStatistics();

        consume += ips.BytesReceived;
        consume += ips.BytesSent;

        IPInterfaceProperties ipp = ni.GetIPProperties();

        consume += ipp.UnicastAddresses.Count;
        consume += ipp.DnsSuffix.Length;
    }

    return consume;
}
```

Reply:
updated.